### PR TITLE
Tools scala3 zio r17

### PR DESCRIPTION
**Pull request by pmeheut**
Attempted solution to: https://github.com/ghostdogpr/caliban/issues/1239

Used https://github.com/scalameta/scalafmt/issues/3035#issuecomment-1007187349 suggestion to load scalafmt dynamically, allowing to generate a Scala 2.12/2.13 & 3 of caliban-tools and using the sbt-codegen plugin in Scala 3.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,48 +1,49 @@
...
 val zioInteropCats2Version = "2.5.1.0"
 val zioInteropCats3Version = "3.2.9.0"
-val zioConfigVersion       = "1.0.10"
-val zqueryVersion          = "0.2.10"
-val zioJsonVersion         = "0.1.5"
-val zioHttpVersion         = "1.0.0.0-RC17"
+val zioConfigVersion = "1.0.10"
+val zqueryVersion = "0.2.10"
+val zioJsonVersion = "0.1.5"
+val zioHttpVersion = "1.0.0.0-RC17"
```
Comment (guizmaii) [inline build.sbt:28]: @pmeheut Please use the project formatter

Comment (pmeheut) [reply build.sbt]: Done.

Comment (guizmaii) [reply build.sbt]: Thanks

**Review by ghostdogpr — APPROVED** [review]
Looks good, just a tiny comment!

**Comment by guizmaii** [comment]
@ghostdogpr @pmeheut Shouldn't we write test to validate that it's working and continue to work over time?

Maybe a new project using Scala3 in the scripted test of the codegen module?

**Comment by pmeheut** [comment]
I can do this if you are in no hurry. Current tests in codegen are Scala 2.12 only I think. Should I make them cross-version or do something specific for Scala 3?

**Comment by guizmaii** [comment]
@pmeheut The best would be to create a third test project, not modifying the two others because they are testing their own stuff.

Sadly, I think we'll be limited by sbt. I remember being forced to use Scala 2.12 instead of 2.13 in the test project because of sbt which is using Scala 2.12 but maybe I was wrong. Maybe I could have used Scala 2.13 which means that you maybe can write this third test project using Scala3.

Can you please give it a try, maybe?

**Comment by pmeheut** [comment]
I'll give it a try and let you know.

**Comment by kyri-petrou** [comment]
Hi there 👋 Are there plans to include this in the next release of caliban-tools by any chance?

**Comment by pmeheut** [comment]
> Hi there 👋 Are there plans to include this in the next release of caliban-tools by any chance?

As soon as my real work stops using all my brain, I'll work on it.

**Comment by jeffmay** [comment]
Is there any plan to release this? I would like to adopt Scala 3 for my Caliban project, but I can't use the code generation tools, which breaks my build.

**Comment by ghostdogpr** [comment]
Closing old PRs, but feel free to reopen it targeting the `series/2.x` branch which is the new default.

**Pull request closed without merging**